airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/DestinationConsumerCallback.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import io.airbyte.commons.lang.CloseableQueue;
+
+public interface DestinationConsumerCallback {
+
+  void queryDatabase(String query) throws Exception;
```

Comment:
do we need queryDatabase at all? if getDatabase returns a `Database` object isn't that enough?

Reply:
I am not convinced yet that we need any of the methods in this interface except `createRawTableQuery`. i suspect we will also need to add `insertIntoTable` or `copyToTable`. As far as I can tell the primitives that are unique to each destinations are:
1. how to create a table
1. how to insert records into a table
1. how to copy records from one table to another table
1. how to truncate a table

i think these are the sorts of things that belong in here. and then instead of calling this `DestinationConsumerCallback` it would be `WriteOperations` or something like that.

Reply:
> do we need queryDatabase at all? if getDatabase returns a Database object isn't that enough?

In the case of JDBC destination, it returns a database but for Snowflake case (which is not a jdbc destination here) or maybe some other destinations, it is using another class of database